### ISSUE TITLE
Google Analytics internal template doesn't interact with cookie consent

### ISSUE THREAD

**Issue by ghalse** [gohugoio/hugo#8262]
In the current implementation of the Google Analytics internal template, there seems to be no way to disable analytics based on user preference through a "cookie consent" interface. These are increasingly common, and it would be nice if there were a way for the internal template to honour a runtime request not to set cookies.

At the moment the template allows for the honouring of a browser Do Not Track preference, and the entire code is wrapped in a `doNotTrack` boolean. That completely prevents the use of analytics for users who set this browser preference, which is fantastic. However, so far as I can tell there's no way to influence the value of this variable from outside the template. And there's (understandably) no way for javascript to toggle the browser DNT preference.

**Comment by djatwood** [comment]
I believe you can turn on `useSessionStorage` to [disable the use of cookies](https://gohugo.io/about/hugo-and-gdpr/#googleanalytics). But I haven't tested this myself.

You should note that Google Analytics v4 doesn't support disabling cookies at all. #8264

**Comment by ghalse** [comment]
Unfortunately, whether you use local storage or cookies doesn't change the "cookie consent" problem, at least not from a privacy law perspective. The underlying directive covers all forms of storage. So this isn't really about disabling cookies so much as allowing users to opt-out of tracking at run time.

Your idea of using localStorage could provide a workaround, but the existing internal template would need to be amended to allow for that since it current (as of the v0.80.0 snap) generates Javascript code that reads:
```
var dnt = (navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack);
var doNotTrack = (dnt == "1" || dnt == "yes");
if (!doNotTrack) {
...
```
which I can't manipulate from outside the template.

**Comment by djatwood** [comment]
I just realized that there is no need for template changes. `doNotTrack` is a global variable that you can modify yourself.
```html
<script>
if (localStorage.getItem("doNotTrack") === null) {
    localStorage.setItem("doNotTrack", "1") // default value before any choice has been made
}
var doNotTrack = localStorage.getItem("doNotTrack") == "1";
</script>
{{ template "_internal/google_analytics.html" . }}
```

Update the localstorage item after consent, reload the page and you should be good to go. This works because `var doNotTrack` is the exact same variable as `window.doNotTrack`.

I've only tested it a little bit but it looks like it should work in Chrome, Safari, and Firefox.

**Comment by ghalse** [comment]
Oh, that's a clever side effect of the choice of variable name, and something I'd not cottoned onto. I guess it also relies on the fact that the `var doNotTrack = (dnt == "1" || dnt == "yes");` assignment doesn't change the existing value of doNotTrack if dnt is undefined.

So maybe the substance of this issue should be to document that since it's non-obvious to people who only occasionally dabble in javascript (I asked several people before opening the issue :)).

**Comment by djatwood** [comment]
I agree that it should be documented. I'll see what I can come up with later this week.
